sonarqube/datadog_checks/sonarqube/check.py

```diff
@@ -0,0 +1,201 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import re
+
+from requests.exceptions import RequestException
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+from .constants import CATEGORIES, NUMERIC_TYPES
+
+
+class SonarqubeCheck(AgentCheck):
+    __NAMESPACE__ = 'sonarqube'
+    SERVICE_CHECK_CONNECT = 'api_access'
+
+    def __init__(self, name, init_config, instances):
+        super(SonarqubeCheck, self).__init__(name, init_config, instances)
+
+        self._web_endpoint = self.instance.get('web_endpoint', '')
+        self._tags = ['endpoint:{}'.format(self._web_endpoint)]
+        self._tags.extend(self.instance.get('tags', []))
+
+        # Construct the component data on the first check run
+        self._components = None
+
+        self.check_initializations.append(self.parse_config)
+
+    def check(self, _):
+        try:
+            self.collect_metrics()
+        except RequestException as e:
+            self.service_check(self.SERVICE_CHECK_CONNECT, self.CRITICAL, tags=self._tags, message=str(e))
+        else:
+            self.service_check(self.SERVICE_CHECK_CONNECT, self.OK, tags=self._tags)
+
+    def collect_metrics(self):
+        available_metrics = self.discover_available_metrics()
+
+        for component, (tag_name, should_collect_metric) in self._components.items():
+            keys_to_query = []
+
+            for key, metric in available_metrics.items():
+                if should_collect_metric(metric):
+                    keys_to_query.append(key)
+
+            if not keys_to_query:
+                self.log.warning('Pattern for component `%s` does not match any available metrics', component)
+
+            response = self.http.get(
```

Comment:
Since you're using the `http` wrapper, you should add `instances/http` and `init_config/http` in `spec.yaml`. I think the last CI run used the old buggy version of `ddev validate http`, so it didn't trigger it.